disnake/mentions.py

```diff
@@ -111,6 +112,22 @@ def none(cls: Type[A]) -> A:
         """
         return cls(everyone=False, users=False, roles=False, replied_user=False)
 
+    @classmethod
+    def from_message(cls: Type[A], message: Message) -> A:
+        """A factory method that returns a :class:`AllowedMentions` dervived from the current :class:`.Message` state.
+
+        Note that this is not what AllowedMentions the message was sent with, but what the message actually mentioned.
+        For example, a message that successfully mentioned everyone will have :attr:`~AllowedMentions.everyone` set to ``True``.
+
+        .. versionadded:: 2.6
+        """
+        return cls(
+            everyone=message.mention_everyone,
+            users=message.mentions.copy(),  # type: ignore # mentions is a list of Snowflakes
+            roles=message.role_mentions.copy(),  # type: ignore # mentions is a list of Snowflakes
+            replied_user=False,
```

Comment:
`replied_user` shouldn't be set to false by default, or alternatively the documentation should mention that caveat. This currently doesn't work, i.e. it no longer mentions the user after editing the message, with `AllowedMentions.none()` set globally:
```py
reply = await ctx.message.reply("h", allowed_mentions=AllowedMentions(replied_user=True))
await asyncio.sleep(1)
await reply.edit("h2", allowed_mentions=AllowedMentions.from_message(reply))
```

I believe the client solves this through setting `replied_user` on edit by checking for `MessageType.reply` and whether the referenced message's author is part of `message.mentions`.

Reply:
Will update. I thought that providing that mention in the list of user mentions would override this.

Reply:
https://github.com/DisnakeDev/disnake/pull/603/commits/b0b2a6d0e34108794e4cb3816373842ace63621a